src/common/middleware/admin.middleware.ts

```diff
@@ -0,0 +1,36 @@
+import passport from 'passport';
+import { Strategy, ExtractJwt } from 'passport-jwt';
+import { Unauthorized } from '../errors/unauthorized';
+import { authModel } from '../../modules/auth/auth.service';
+
+const customFields = {
+    jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    secretOrKey: 'secret',
+    passReqToCallback: true
+}
+
+const strategy = new Strategy(customFields, 
+    async function (req,jwtPayload, done) {
+
+        try {
+
+            const user = await authModel.findUserById(jwtPayload.id);
+            if (!user) {
+                done (null,false)
+            } 
+            else if (user.role !== 'admin') {
+                done(null,false);
+            }
+            else {
+                req.params.user = user;
+                done(null,user);
+            }
+        } catch(err) {
```

Comment:
you dont need all these code at all. dont use passport here

just use if (res.locals.user === 'admin') ...